File: src/Topics/listener.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"

// CONSTANTS
const char *NODE_NAME {"listener_cpp"};
const char *TOPIC_NAME {"chat"};

// GLOBALS
ros::Subscriber sub;

// chat subscriber callback
void callback(const std_msgs::String::ConstPtr& msg){
    ROS_INFO("[%s] I heard [%s]", NODE_NAME, msg->data.c_str());
    return;
}

// setting up this node
inline void setup(int argc, char **argv){
    ros::init(argc, argv, NODE_NAME);
    static ros::NodeHandle node;
    // static cause it's important that NodeHandle ojbect lives for as long as program is running

    // setting up subscriber
    sub = node.subscribe(TOPIC_NAME, 10, callback);
    return;
}

// main driver
int main(int argc, char **argv){
    setup(argc, argv);
    ros::spin();    // keep the program from exiting unless terminated by user
    return 0;
}

File: src/Topics/listener.py
#! /usr/bin/env python

import rospy
from std_msgs.msg import String

# CONSTANTS
NODE_NAME = "listener_py"
TOPIC_NAME = "chat"

# GLOBALS
sub = None      # Will be subscriber object for topic name

# callback function for /chat topic
def callback(msg):
    rospy.loginfo(f"I heard [{msg.data}]")

# setup node
def setup():
    global sub
    rospy.init_node(NODE_NAME)
    sub = rospy.Subscriber(TOPIC_NAME, String, callback, queue_size=10)


if __name__ == "__main__":
    try:
        setup()
        rospy.spin()    # keep node from exiting and allow ros to process incoming msgs
    except rospy.ROSInterruptException:
        pass